Topic5/src/Demo.java

```diff
@@ -0,0 +1,131 @@
+import java.sql.SQLException;
+import java.util.Scanner;
+
+public class Demo {
+
+	public static void main(String[] args) {
+		
+		java.sql.Connection connection;
+		try {
+			Class.forName("com.mysql.jdbc.Driver").newInstance();
+
+			// connection to the DB
+			String servidor = "localhost:3306";
+			String baseDatos = "high_school";
+			String uriConexion = "jdbc:mysql://" + servidor + "/" + baseDatos;
+			String user = "adminBD";
+			String password = "admin";
+
+			connection = java.sql.DriverManager.getConnection(uriConexion, user, password);
+
+			java.sql.Statement statement = connection.createStatement();
+
+			// KEY POINT 4
+			System.out.println("List students and teachers for a given course.");
+			System.out.println("Courses: ");
+			System.out.println("1.Java course 2.PHP course 3.C course 4.C++ course 5.Phyton course");
+			System.out.print("Choose a course number: ");
+
+			// the id_course chosen by the user
+			int id_course;
+
+			Scanner scanner = new Scanner(System.in);
+			id_course = scanner.nextInt();
+
+			// the id_course must be between 1 and 5
+			while (id_course < 1 || id_course > 5) {
+				System.out.println();
+				System.out.println("Wrong choice, please choose one of the following number course: ");
+				System.out.println("1.Java course 2.PHP course 3.C course 4.C++ course 5.Phyton course");
+				System.out.print("Choose a course number: ");
+				id_course = scanner.nextInt();
+			}
+
+			System.out.println();
+
+			// query to obtain the name of the course and the first_name and
+			// last_name of the teacher
+			String sql = "SELECT name_course,first_name,last_name FROM Course NATURAL JOIN Teacher WHERE id_course=" + id_course;
+
+			java.sql.ResultSet resultSet = statement.executeQuery(sql);
+			resultSet.next();
+			System.out.println("Course: " + resultSet.getString(1));
+			System.out.println("Teacher: " + resultSet.getString(3) + ", " + resultSet.getString(2));
+			System.out.println("Students: ");
+
+			// query to obtain the first_name and last_name from students in the
+			// chosen course ordered by alphabetically by last_name
+			sql = "SELECT first_name,last_name FROM Student_Course NATURAL JOIN Student WHERE id_course=" + id_course
+					+ " ORDER BY last_name";
+
+			resultSet = statement.executeQuery(sql);
+
+			// print the students of the course
+			while (resultSet.next()) {
+				System.out.println("\t " + resultSet.getString(2) + ", " + resultSet.getString(1));
+			}
+
+			System.out.println();

```

Comment:
I mean to use a single (and complex) query to get the same result that you are having with two separate queries and some processing, do you believe that in this case it would be possible?